packages/fxa-settings/src/lib/utilities.ts

```diff
@@ -35,6 +36,16 @@ export function deepMerge(
   return deepMerge(target, ...sources);
 }
 
+export function getSearchParams(paramKeys: string[], locationHref: string) {
+  const { searchParams } = new URL(locationHref);
```

Comment:
Will `locationHref` always be an absolute URL?
[`URL`](https://developer.mozilla.org/en-US/docs/Web/API/URL/URL) throws hard errors if we don't have a domain part; unless we specify a base URL. Otherwise we might be able to just use [`URLSearchParams`](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/URLSearchParams).

Reply:
Unfortunately, we cannot modify `url.search` in Storybook, which using `URLSearchParams` would require us to do. I spent a few hours trying to get it to work, tried hacky solutions using `@ts-ignore` and trying to reset it via `Object.defineProperty`, and nothing let me force-set it. 😩

Reply:
> Unfortunately, we cannot modify url.search in Storybook

In my password reset PR, I was able to pass a custom history objection into the `LocationProvider` and set the url that way. Would something like that help in this case?

https://github.com/mozilla/fxa/blob/30a8adf8d34ae29dfc1e2ea51d2f10fa45aef1be/packages/fxa-settings/src/pages/ResetPassword/CompleteResetPassword/index.stories.tsx#L38

Reply:
@vbudhram I actually ended up copying what you did. 😁 We can set `href` just fine, but can't set `search`.